nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/RestLookupService.java

```diff
@@ -125,11 +127,30 @@ public class RestLookupService extends AbstractControllerService implements Reco
         .identifiesControllerService(SSLContextService.class)
         .build();
 
+    public static final PropertyDescriptor AUTHENTICATION_STRATEGY = new PropertyDescriptor.Builder()
+            .name("rest-lookup-authentication-strategy")
+            .displayName("Authentication Strategy")
+            .description("Authentication strategy to use with REST service.")
+            .required(true)
+            .allowableValues(AuthenticationStrategy.class)
+            .defaultValue(AuthenticationStrategy.NONE)
+            .build();
+
+    public static final PropertyDescriptor OAUTH2_ACCESS_TOKEN_PROVIDER = new PropertyDescriptor.Builder()
+        .name("rest-lookup-oauth2-access-token-provider")
+        .displayName("OAuth2 Access Token Provider")
+        .description("Enables managed retrieval of OAuth2 Bearer Token applied to HTTP requests using the Authorization Header.")
+        .identifiesControllerService(OAuth2AccessTokenProvider.class)
+        .required(false)
```

Comment:
This can be changed to `true` since the `dependOn()` check makes it conditional, and it is required when the associated strategy is selected.
```suggestion
        .required(true)
```

Reply:
@exceptionfactory I assume this will also apply to the basic auth username and password fields as well now that they also depend on the selection.

Reply:
That could apply to those properties, but since they are existing, it seems reasonable to leave them unchanged for the scope of this particular pull request so as not to introduce any unexpected changes.